From cde09c892e43c43aea42224c709c7ecc0fe0f1c9 Mon Sep 17 00:00:00 2001
From: Soroosh Haeri <soroosh.haeri@icloud.com>
Date: Sun, 16 Aug 2015 07:06:37 -0700
Subject: [PATCH] There is a starnge error with istingstream using clang (when
 the code is compiled with gnu g++ istringstream does not fail). It fails when
 the quantity class tries to extract its value and units from an input string.
 I changed this to use std::atod instead of istringstream to fix the problem.

---
 cpp/src/quantity.cpp | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/cpp/src/quantity.cpp b/cpp/src/quantity.cpp
index 352fad0..488f0f9 100644
--- a/cpp/src/quantity.cpp
+++ b/cpp/src/quantity.cpp
@@ -20,11 +20,13 @@ Quantity::Quantity(const MeasurementUnit &converter_) :
 	value(0), unit(converter_.getBaseUnit()), converter(converter_) {}
 
 void Quantity::fromString(const std::string &str) {
-	std::istringstream ss(str);
-	ss>>this->value>>this->unit;
+        std::string::size_type sz;
+        this->value = std::stod (str,&sz);
+        this->unit = str.substr(sz);
+        this->unit.erase(remove_if(this->unit.begin(), this->unit.end(), isspace), this->unit.end());
 
-	if(this->unit == "")
-		this->unit = this->converter.getBaseUnit();
+        if(this->unit == "")
+                this->unit = this->converter.getBaseUnit();
 }
 
 std::string Quantity::toString(const std::string &separator) const {
-- 
2.3.2 (Apple Git-55)

